src/csv_parser.c

```diff
@@ -0,0 +1,309 @@
+ * \version 1.0
+ * \date 2019-Dec-14
+ */
+
+#include <stdio.h>
+#include <conio.h>
+#include <stdlib.h>
+#include <string.h>
+
+/** Define the number of columns as 3 for storing CSV file into a 2D Array. 
+ *  Input CSV file has 3 Columns Time Stamp, Sensor name, Sensor Reading */
+	
+#define COL 3 
+
+/**
+ * \brief Helper Function to get Sample CSV File Size
+ * \details Use of File stream and File functions fseek and ftell to 
+ *  compute the size
+ * \param[in] Pointer to the File of Input CSV
+ * \return Integer: Value representing Size of the file
+ * \return Failure: No Error messages
+ */
+ 
+int getfilesize(char *fname){
+  /*  Open the stream pointer to the file passed by the function call
+   * and set cursor to end and count by using ftell
+   */
+	FILE *fp = fopen(fname, "r"); 
+    fseek(fp, 0, SEEK_END);
+    int size = ftell(fp);
+    fclose(fp);
+    return size;
+}
+
+/**
+ * \brief Reads csv file into a local buffer
+ * \details Tokenize string at comma, convert text data to integer and 
+ *  write it into 2D buffer
+ * \param[in] Pointer to the File of Input CSV, Pre-Initialized 2D Array
+ *  named csv[Number of Columns], Integer with number of Rows
+ * \return Integer: 2D Array with rowsx3 matrix where rows is equal to 
+ *	number of time stamps read from Input CSV
+ * \return Failure: No Error message built into it
+ */
+ 
+void read_csv(char *fname, float (*array)[COL], int rows){
+    /* allocate memory buffer for file read 
+    *  read csv file into buffer 
+    */
+    int size = getfilesize(fname);
+    char *buf = malloc(size +1);
+    FILE *fp = fopen(fname, "r");
+    fread(buf, size, 1, fp);
+    fclose(fp) ;
+    /* structurized tokenize for comma or new line */
+    char *tok = strtok(buf, ",\n");
+    for(int i=0;i<rows;i++){
+    	for(int j=0;j<COL;j++){
+      		if(tok == NULL){
+       			break;
+       		}
+      		/* convert text numeral to integer */
+      		float val = atof(tok);
+      		/* Store each fields of CSV file into array[][] */
+      		array[i][j] = val;
+ 			/* tokenize at comma or newline */
+     		tok = strtok(NULL, ",\n");
+     	}
+    }
+	free(buf);
+}
+
+/**
+ * \brief main function, accepts input CSV file and calls csv helper functions
+ *  to obtain 2D array and extract sensor reading for each time stamp.
+ * \details Call readcsv() function to parse CSV file in the form 
+ *  of 2D array and perform array operations to extract sensor readings
+ *  for each time stamp and store it in a 2D array
+ * \param[in] CSV File
+ * \return 2D array: csv[][] [rowsx3] matrix where rows is equal to 
+ *  number of time stamp read from Input CSV
+ * \return 2D array: sr[][] with all sensor readings for each time stamp
+ * \return 1D array: u_ts[] position of unique time stamps in CSV file starting
+     from 2nd Time stamp
+ * \return 1D array: sr_c[] number of sensors in each individual time stamps 
+ * \return Failure: No Error messages
+ */
+ 
+int main()
```

Comment:
Generally you should only have the single main function in main.c. Giving this a different name would be help clear up any confusion when it comes around to producing documentation for developers.

Reply:
Noted, Fixed changes in the latest dev branch pull request